components/definition/cellRules/sections/mechanisms/NetPyNEMechanism.js

```diff
@@ -1,107 +1,65 @@
+import NetPyNEField from '../../../../general/NetPyNEField';
 
 var PythonControlledCapability = require('../../../../../../../js/communication/geppettoJupyter/PythonControlledCapability');
 var PythonControlledTextField = PythonControlledCapability.createPythonControlledControl(TextField);
-var PythonControlledSelectField = PythonControlledCapability.createPythonControlledControl(SelectField);
+var PythonMethodControlledSelectField = PythonControlledCapability.createPythonControlledControlWithPythonDataFetch(SelectField);
 
 export default class NetPyNEMechanism extends React.Component {
 
   constructor(props) {
     super(props);
-    var _this = this;
     this.state = {
-      currentName: props.name,
-      selectedIndex: 0,
-      sectionId: "General"
+      currentMech: props.name,
+      mechFields: ''
     };
-
-
-    this.setPage = this.setPage.bind(this);
-  }
-
-  setPage(page) {
-    this.setState({ page: page });
-  }
-
-  select = (index, sectionId) => this.setState({ selectedIndex: index, sectionId: sectionId });
-
-  handleRenameChange = (event) => {
-    var that = this;
-    var storedValue = this.props.name;
-    var newValue = event.target.value;
-    this.setState({ currentName: newValue });
-    this.triggerUpdate(function () {
-      // Rename the population in Python
-      Utils.renameKey("netParams.cellParams['" + that.props.cellRule + "']['secs']['" + that.props.section + "']['mechs']", storedValue, newValue, (response, newValue) => { });
-    });
-
-  }
-
-  triggerUpdate(updateMethod) {
-    //common strategy when triggering processing of a value change, delay it, every time there is a change we reset
-    if (this.updateTimer != undefined) {
-      clearTimeout(this.updateTimer);
-    }
-    this.updateTimer = setTimeout(updateMethod, 500);
-  }
-
-  getBottomNavigationItem(index, sectionId, label, icon) {
-
-    return <BottomNavigationItem
-      key={sectionId}
-      label={label}
-      icon={(<FontIcon className={"fa " + icon}></FontIcon>)}
-      onClick={() => this.select(index, sectionId)}
-    />
-  }
+    this.findmechFields = this.findMechFields.bind(this);
+    this.renderMechFields = this.renderMechFields.bind(this);
+  };
+  
   componentWillReceiveProps(nextProps) {
-    this.setState({ currentName: nextProps.name});
-  }
+    this.setState({ currentMech: nextProps.name});
+  };
+  
+  findMechFields = () => {
+    Utils
+      .sendPythonMessage("netpyne_geppetto.getMechParams", [this.state.currentMech])
+      .then((response) => {
+        if (JSON.stringify(this.state.mechFields)!=JSON.stringify(response))
+        this.setState({mechFields: response})
+      })
+  };
+  
+  renderMechFields = () => {
+    if (this.state.mechFields=='') return <div key={"empty"}/>
+    var tag = "netParams.cellParams['" + this.props.cellRule + "']['secs']['" + this.props.section + "']['mechs']['" + this.state.currentMech + "']"
+    return this.state.mechFields.map((name, i) =>
```

Comment:
[This method over here](https://github.com/MetaCell/geppetto-netpyne/blob/development/components/definition/populations/NetPyNEPopulation.js#L44-L62) does sth similar to what you are trying to do here. Code on the python side can be found in api.py. I think there are some good ideas (and probably bad ones) we can try to merge. What do you think about merging the two approaches and improve a bit this part (pass a proper metadata structure that potentially provides help, hint, type, etc and remove redundancy)?